components/locid_transform/src/expander.rs

```diff
@@ -507,11 +507,51 @@ impl LocaleExpander {
     /// assert_eq!(lc.minimize(&mut locale), TransformResult::Unmodified);
     /// assert_eq!(locale, locale!("zh"));
     /// ```
-    pub fn minimize<T: AsMut<LanguageIdentifier>>(&self, mut langid: T) -> TransformResult {
+    pub fn minimize<T: AsMut<LanguageIdentifier>>(&self, langid: T) -> TransformResult {
+        self.minimize_favor_region(langid, true)
+    }
+
+    /// This returns a new Locale that is the result of running the
+    /// 'Remove Likely Subtags, favoring script' algorithm from
+    /// <https://www.unicode.org/reports/tr35/#Likely_Subtags>.
+    ///
+    /// If the result of running the algorithm would result in a new locale, the
+    /// locale argument is updated in place to match the result, and the method
+    /// returns [`TransformResult::Modified`]. Otherwise, the method
+    /// returns [`TransformResult::Unmodified`] and the locale argument is
+    /// unchanged.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use icu_locid::locale;
+    /// use icu_locid_transform::{
+    ///     LocaleExpander, TransformResult
+    /// };
+    ///
+    /// let lc = LocaleExpander::new();
+    ///
+    /// let mut locale = locale!("zh_TW");
+    /// assert_eq!(lc.minimize_favor_script(&mut locale), TransformResult::Modified);
+    /// assert_eq!(locale, locale!("zh_Hant"));
+    /// ```
+    pub fn minimize_favor_script<T: AsMut<LanguageIdentifier>>(
+        &self,
+        langid: T,
+    ) -> TransformResult {
+        self.minimize_favor_region(langid, false)
+    }
+
+    fn minimize_favor_region<T: AsMut<LanguageIdentifier>>(
+        &self,
+        mut langid: T,
+        favor_region: bool,
```

Comment:
Alternatives to the the `favor_region` argument:
- Duplicate code across `minimize` and `minimize_favor_script`.
- Define an enum for option-passing instead of a bool. (something like `self.minimize_favor(MinimizePriority::Region)`
- Define consts for the bool to make it read better (`const FAVOR_REGION = true`, `self.minimize_favor(FAVOR_REGION)`

Currently used `bool` since it's the simplest and shortest.

Reply:
Suggestion: I'd probably call it minimize_impl or minimize_internal